### ISSUE TITLE
Cannot seem to locally stup turnserver.

### ISSUE THREAD

**Issue by pwninitd** [pion/turn#399]
I am very new to this sphere and I just wanted to apologies if I take too much of your guys is time and if this isn't the appropriate place to ask this question (if so please redirect me I will ask my question elsewhere), but for now I don't know where to turn to for questions.

I tried to setup the turn-server/log example on my local machine but with no success. When I try to authenticate using trickle ICE I cannot seem to get any results.

This is what I get on trickle ICE
![Screenshot from 2024-06-23 14-03-46](https://github.com/pion/turn/assets/48697165/5fc49d6f-6508-4e97-b395-1bed418c5a04)

And this is the log I get from the turn server
![Screenshot from 2024-06-23 14-05-13](https://github.com/pion/turn/assets/48697165/e7695385-200a-4004-9ccd-3d43b3ecc266)

And I use this command to start it ./log -public-ip 127.0.0.1 -users "username=password"
If anyone can help me out I will be very grateful

**Comment by rg0now** [comment]
The weird fact about the TURN protocol is that the first packet sent by the client to the server will by design fail with a TURN "unauthenticated" error: this is how the TURN server returns a nonce for the client, which is in turn the main protection in TURN against replay attacks. So the client sends a request without a nonce, the server replies with an error but includes the nonce in the response, and from this point the client should retry with a nonce, this is what's not happening here for some reason. My totally uneducated guess would be that the browser WebRTC protocol stack contains a limitation that it stops gathering relay candidates from local TURN servers, but this is just a totally arbitrary guess. Just deploy the TURN server to a public IP (or a Docker container) and it will just work. 

TL;DR The TURN error is by design, and the reason for the weirdness is on the browser side but I don't know what.

**Comment by pwninitd** [comment]
I've read something about the authentication and I know that the first response is a 401, but what confused me is that there were multiple of them. Anyways thank you for your response, however I have one more question. Can you guide me to any resources for deploying a turn server on a docker container?

**Comment by rg0now** [comment]
>  what confused me is that there were multiple of them

This is because the browser retries several times.

> Can you guide me to any resources for deploying a turn server on a docker container?

[Here](https://github.com/pion/turn/blob/master/examples/turn-server/simple/Dockerfile) is a prefab Dockerfile for the "simple" TURN server example in pion/turn. There are also a bunch of complete TURN server images available at popular container image repositories.

Closing this for now. Feel free to reopen if you find out the reason why the ICE test fails.
